media/js/base/mozilla-client.js

```diff
@@ -321,72 +322,113 @@ if (typeof Mozilla === 'undefined') {
+     *     - desktopDevices & mobileDevices
+     *          - if logged in and Fx >= 50: the number of linked devices
+     *          - if logged in and Fx < 50 && Fx > 29: 'unknown'
+     *          - if logged out or Fx < 29 or not Fx: false
+     * Notes:
+     * - Fx < 50 has FxA and UITour support but the API does not return device counts
+     * - Fx < FxALastSupported accounts.firefox.com does not work
+     *     - FxALastSupported is supplied by the FxA team
+     *     - these versions are still capable of logging in through the browser
+     *     - differentiated because we generally do not give these versions the FxA calls to action (eg. "Create an Account")
+     * - Fx < 29 the mozUITour API is not available, though the user may still be logged in
+     * - If you're curious, "sync" began with Fx 4.
+     *
+     * @param  {Function} callback - callback function to be executed with the FxA details
      * @return {None}
      */
-    Client.getSyncDetails = function (callback) {
-        // Fire the callback function immediately if cache exists
-        if (Client.SyncDetails) {
-            callback(Client.SyncDetails);
+    Client.getFxaDetails = function (callback) {
+        // Fire the callback function immediately if FxaDetails are already defined
+        if (Client.FxaDetails) {
+            callback(Client.FxaDetails);
 
             return;
         }
-
-        var callbackID = Math.random().toString(36).replace(/[^a-z]+/g, '');
-
-        var listener = function (event) {
-            if (!event.detail || !event.detail.data || event.detail.callbackID !== callbackID) {
-                return;
-            }
-
-            window.clearTimeout(timer);
-
-            // device counts are only available in Fx50+, so provide fallback values in case
-            onRetrieved(event.detail.data.setup, event.detail.data.desktopDevices || 0, event.detail.data.mobileDevices || 0, event.detail.data.totalDevices || 0);
+        // set up the object with default values of false
+        var details = Client.FxaDetails = {
+            'firefox': false,
+            'legacy': false,
+            'mobile': false,
+            'setup': false,
+            'desktopDevices': false,
+            'mobileDevices': false
         };
 
-        var onRetrieved = function (setup, desktopDevices, mobileDevices, totalDevices) {
-            document.removeEventListener('mozUITourResponse', listener, false);
-
-            var details = Client.SyncDetails = {
-                'setup': setup,
-                'desktopDevices': desktopDevices,
-                'mobileDevices': mobileDevices,
-                'totalDevices': totalDevices
-            };
+        // override object values as we get more information
+        if(Client.isFirefoxAndroid) {
+            details.firefox = true;
+            details.mobile = 'android';
+        } else if (Client.isFirefoxiOS) {
+            details.firefox = true;
+            details.mobile = 'ios';
```

Comment:
Might be worth calling `returnFxaDetails` in the case of Android/iOS so we don't have to wait for the timer?